### PR TITLE
cpu/stm32f4discovery: fixed freq return in pwm_init

### LINKED ISSUE #4439: stm32f4: pwm_init returns wrong value
In *cpu/stm32f4/periph/pwm.c* `pwm_init` is returning the requested input value.
It should return a frequency that is calculated based on actual clock and prescaler values.

### PR DESCRIPTION
**Pull request by haukepetersen**
fixes #4439

### CODE REVIEW

File: cpu/stm32f4/periph/pwm.c [hunk 1]
```diff
@@ -120,6 +120,8 @@ int pwm_init(pwm_t dev, pwm_mode_t mode, unsigned int frequency, unsigned int re
     }
     tim->PSC = (pwm_clk / (resolution * frequency)) - 1;
     tim->ARR = resolution - 1;
+    /* calculate the actual PWM frequency */
+    frequency = (pwm_clk / (resolution * tim->PSC));
```
Comment (A-Paul) [inline cpu/stm32f4/periph/pwm.c]: (tim->PSC + 1) to compensate for the "one" you subtracted in line 121. ;-)

Comment (haukepetersen) [inline cpu/stm32f4/periph/pwm.c]: makes sense

**Comment by haukepetersen** [comment]
fixed and squashed

File: cpu/stm32f4/periph/pwm.c [hunk 2]
```diff
@@ -120,6 +120,8 @@ int pwm_init(pwm_t dev, pwm_mode_t mode, unsigned int frequency, unsigned int re
     }
     tim->PSC = (pwm_clk / (resolution * frequency)) - 1;
     tim->ARR = resolution - 1;
+    /* calculate the actual PWM frequency */
+    frequency = (pwm_clk / (resolution * (tim->PSC + 1));
```
Comment (A-Paul) [inline cpu/stm32f4/periph/pwm.c]: Closing parethesis missing.

**Comment by A-Paul** [comment]
Accidently commented in commit. Deleted.

File: cpu/stm32f4/periph/pwm.c (hunk 2 above)
Comment (haukepetersen) [inline cpu/stm32f4/periph/pwm.c]: ups, forgot to press the save button before commiting... fixed

**Comment by A-Paul** [comment]
Thanks for corrections. ACK when Travis is happy.

**Comment by A-Paul** [comment]
Go!

**Comment by OlegHahm** [comment]
Should this also go into 2015.12-branch?

**Comment by A-Paul** [comment]
I thought *2015.12* isn't freezed yet. Will it cause any inconvenience?

**Comment by haukepetersen** [comment]
I think it's rather minor and not really worth the effort, having it for 2016.03 should be fine... (expect @A-Paul disagrees with me...)

**Comment by A-Paul** [comment]
> I think it's rather minor and not really worth the effort...

I dont want to push anything. I simply try to understand "what is the effort?". I also can't find info about the 'backporting-needed' label.

**Comment by OlegHahm** [comment]
The label `backporting-needed` means that backporting to the bugfixing release branch is needed. Will add it to the corresponding Wiki page.

Current state is:
* we are in *feature* freeze for the release (`2015.12`)
* the release will be tagged on the branch `2015.12-branch`
* hence, new features may be merged into `master`
* `2015.12-branch` will only receive bugfixes until feature freeze for `2016.03`

I have no strong opinion regarding this PR, but just cherry-picking this onto `2015.12-branch` and opening a PR with the `Ready for CI build` label set right from the start is definitely less effort than any further discussion. ;)

**Comment by OlegHahm** [comment]
P.S. Thanks, @A-Paul, for pointing out that this label is undocumented.

**Comment by A-Paul** [comment]
@haukepetersen seems like I stumbled into unknown cliffs of the release process.
I have no timing preference. So please, before I mess up things any further, chose what is the most convenient advance.

**Pull request merged**